blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ScheduleContext.java

```diff
@@ -252,17 +264,23 @@ void scheduleEligibleSlots(long maxAgeMillis, long delayedMetricsMaxAgeMillis) {
         }
     }
 
-    private boolean areChildKeysOrSelfKeyScheduledOrRunning(SlotKey slotKey) {
+    boolean areChildKeysOrSelfKeyScheduledOrRunning(SlotKey slotKey) {
         // if any ineligible (children and self) keys are running or scheduled to run, we shouldn't work on this.
         Collection<SlotKey> ineligibleKeys = slotKey.getChildrenKeys();
 
-        if (runningSlots.keySet().contains(slotKey) || scheduledSlots.contains(slotKey)) {
+        if (runningSlots.keySet().contains(slotKey)) {
+            return true;
+        }
+        if (scheduledSlots.contains(slotKey)) {
             return true;
         }
 
         // if any ineligible keys are running or scheduled to run, do not schedule this key.
         for (SlotKey childrenKey : ineligibleKeys) {
-            if (runningSlots.keySet().contains(childrenKey) || scheduledSlots.contains(childrenKey)) {
+            if (runningSlots.keySet().contains(childrenKey)) {
+                return true;
+            }
+            if (scheduledSlots.contains(childrenKey)) {
```

Comment:
@usnavi It has to do with the way jacoco calculates branch coverage. With both conditions in the same if-statement, with a `||` between them, there are theoretically four possible setups: `(T || T)`, `(T || F)`, `(F || T)`, `(F || F)`. The first one is not possible, because a slot should never be both scheduled and running. By breaking up the conditionals into two independent branches, they can be tracked by jacoco separately. Each one says "all of 2 branches covered" instead of "1 of 4 branches missed" in the combined case.